src/helpers/glpTEST/getGlpSwapData.js

```diff
@@ -0,0 +1,318 @@
+
+  const gmxLensContract = await new ethers.Contract(
+    gmxLensAddress,
+    JSON.stringify(gmxLensAbi),
+    provider
+  );
+
+  const tokensArr = await getTokens(provider, gmxLensContract);
+
+  for (let token of tokensArr) {
+    const { buyAmount, data } = await swap0xRequest(
+      chainId,
+      token.address,
+      borrowToken.address,
+      slipage,
+      sellAmount
+    );
+
+    const resp = await gmxLensContract.getMintedGlpFromTokenIn(
+      token.address,
+      buyAmount
+    );
+
+    const getSwapDataEncode = ethers.utils.defaultAbiCoder.encode(
+      ["bytes", "address"],
+      [data, token.address]
+    );
+
+    results.push({
+      ...token,
+      tokenName: getNameFromAddress(token.address),
+      token: token.address,
+      glpOutAmount: resp[0].toString(),
+      feesPoints: resp[1].toString(),
+      swapDataEncode: getSwapDataEncode,
+    });
+
+    store.commit("updateRouteData", results);
+  }
+
+  const tokenInfoArray = results.sort(maxGlpAmount);
+
+  let globalLimit = ethers.BigNumber.from(0);
+  tokenInfoArray.forEach((item) => {
+    globalLimit = globalLimit.add(item.maxAmountInToMIM);
+  });
+  console.log("globalLimit", globalLimit.toString());
+
+  if (sellAmount.gt(globalLimit)) console.log("to much mim to swap"); // TODO: add notification
+
+  let mimLeftToSwap = sellAmount;
+  const cookInfo = [];
+
+  for (let info of tokenInfoArray) {
+    if (mimLeftToSwap.eq(0)) break;
+    console.log("info", info);
+    console.log("mimLeftToSwap", mimLeftToSwap.toString());
+    console.log("maxAmountInToMIM", info.maxAmountInToMIM.toString());
+    const itsAmountEnough =
+      info.maxAmountInToMIM.gt(mimLeftToSwap) ||
+      info.maxAmountInToMIM.eq(mimLeftToSwap);
+    const awailableToSwap = itsAmountEnough
+      ? mimLeftToSwap
+      : info.maxAmountInToMIM;
+
+    // add cook call info block
+    if (itsAmountEnough && cookInfo.length === 0) {
+      info.sellAmount = awailableToSwap;
+      cookInfo.push(info);
+    } else {
+      const { data } = await swap0xRequest(
+        chainId,
+        info.address,
+        borrowToken.address,
+        slipage,
+        awailableToSwap
+      );
+
+      const getSwapDataEncode = ethers.utils.defaultAbiCoder.encode(
+        ["bytes", "address"],
+        [data, info.address]
+      );
+
+      info.sellAmount = awailableToSwap;
+      info.swapDataEncode = getSwapDataEncode;
+
+      cookInfo.push(info);
+    }
+    //
+
+    mimLeftToSwap = mimLeftToSwap.sub(awailableToSwap);
+  }
+
+  console.log("cookInfo", cookInfo);
+  console.log("result", tokenInfoArray);
+
+  for (let info of cookInfo) {
+    const oracleExchangeRate = pool.oracleExchangeRate;
+
+    const slippageBN = ethers.BigNumber.from(
+      parseFloat(slipage * 1e10).toFixed(0)
+    );
+
+    const userAddr = store.getters.getAccount;
+    const swapAmount = info.sellAmount;
+
+    const expectedCollateral = swapAmount
+      .mul(oracleExchangeRate)
```

Comment:
why don't we reuse glpOutAmount that we have computed for each of the tokens in L139 as this will have greater accuracy?